public/script.js

```diff
@@ -1,74 +1,177 @@
+  })
+  var profile = JSON.parse(window.localStorage.getItem('profile'))
+  var username = profile.nickname
+  var msg = document.querySelector('#msg')
 
-  section.innerHTML = `
-  <p>${message.username}</p>
-  <p>${message.message}</p>
-  <p>${time}</p>`
+  if (!username) {
+    var ran = Math.floor((Math.random() * 999) + 100)
+    username = 'user' + ran
+  }
 
-  chat.appendChild(section)
-  section.scrollIntoView()
-}
+  document.querySelector('#username').textContent = username
 
-// function checkMessage () {
-//   if (event.keyCode !== 13) return
-//   if (event.)
-// }
-
-function sendMessageToServer (event) {
-  console.log(event)
-  if (event.type === 'keydown' && event.keyCode !== 13) return
-  if (msg.value === '') return
-  var timestamp = new Date()
-  var message = {
-    username: username,
-    message: msg.value,
-    timestamp: timestamp.toISOString()
+  function getElapsedTime (timestamp) {
+    var ms = Date.now() - timestamp.getTime()
+    var min = ms / 60000
+    var time = ''
+    if (min > 1439) {
+      time = `${Math.round(min / 1440)}d`
+    } else if (min > 59) {
+      time = `${Math.round(min / 60)}h`
+    } else if (min >= 1) {
+      time = `${Math.round(min)}m`
+    } else if (min > 0) {
+      var sec = Math.round(min * 60)
+      if (sec === 0) time = 'just now'
+      else time = `${sec}s`
+    }
+    return time
   }
-  socket.emit('chat message', message);
-  msg.value = ''
-}
 
-var btn = document.querySelector('#btn')
-btn.addEventListener('click', sendMessageToServer, false)
+  function addMessageToList (message) {
+    var chat = document.querySelector('.chat')
+    var section = document.createElement('section')
 
-msg.addEventListener('keydown', sendMessageToServer, false)
+    var timestamp = new Date(message.timestamp)
+    var time = getElapsedTime(timestamp)
 
-socket.on('chat message', addMessageToList)
+    section.innerHTML = `
+    <p>${message.username}</p>
+    <p>${message.message}</p>
+    <p>${time}</p>`
 
-socket.on('chat log', function (messages) {
-  messages.forEach(addMessageToList)
-})
+    chat.appendChild(section)
+    section.scrollIntoView()
+  }
+
+  // function checkMessage () {
+  //   if (event.keyCode !== 13) return
+  //   if (event.)
+  // }
+
+  function sendMessageToServer (event) {
+    console.log(event)
+    if (event.type === 'keydown' && event.keyCode !== 13) return
+    if (msg.value === '') return
+    var timestamp = new Date()
+    var message = {
+      username: username,
+      message: msg.value,
+      timestamp: timestamp.toISOString()
+    }
+    socket.emit('chat message', message)
+    msg.value = ''
+  }
+
+  var btn = document.querySelector('#btn')
+  btn.addEventListener('click', sendMessageToServer, false)
+
+  msg.addEventListener('keydown', sendMessageToServer, false)
+
+  socket.on('chat message', function (message) {
+    addMessageToList(message)
+    $('.typingNotification').remove()
```

Comment:
I removed jQuery in my last build, so this won't work. Are you able to refactor it to not use jQuery?